chore: add structured package data for packages in `math/base/special`

gunjjoshi opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: na
  - task: lint_package_json status: passed
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: na
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: na
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

Resolves a part of #7924.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds structured package data for the following packages:
    - [`math/base/special/acscf`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/acscf)
    - [`math/base/special/acsch`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/acsh)
    - [`math/base/special/asin`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/asin)
    - [`math/base/special/asinf`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/asinf)

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #7924.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gunjjoshi committed:
chore: add scaffold metadata

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/math/base/special/acsc/package.json:
```diff
@@ -66,4 +66,78 @@
+            ]
+          },
+          "example_values": [
+            -5.0,
```
In general, you need to generate at least `20` example values.

gunjjoshi committed:
docs: add more examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gunjjoshi committed:
Merge branch 'develop' into metadata

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi commented:
I've now added more examples, and removed `acsc` from this PR, as the package data for `acsc` was already added via https://github.com/stdlib-js/stdlib/pull/8026

gunjjoshi commented:
The CI errors state that:
```
ERROR: package.json in lib/node_modules/@stdlib/math/base/special/asin needs updates to directories and/or gypfile fields
```
But I see all directories already present in the `directories` filed, and the `gypfile` field set to `true`, for example, in [`asin`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/asin/package.json#L17).

Planeshifter commented:
@gunjjoshi 

The workflow failed because it internally serializes and deserializes  the JSON blobs, which causes the trailing zeros to be removed. See the diffs, e.g.:

```
-            1.0
+            1
```

It's on my TODO list to investigate this problem with the check for changes in the `package.json` files and find a solution to not fail when neither `gypfile` or `directories` is incorrect. For the time being, I propose removing the trailing zeros for tests to pass.

kgryte commented on lib/node_modules/@stdlib/math/base/special/acscf/package.json:
```diff
@@ -66,4 +66,89 @@
+            ]
+          },
+          "example_values": [
+            -5.0,
+            -4.2,
+            -3.7,
+            -3.1,
+            -2.6,
+            -2.2,
+            -1.7,
+            -1.3,
+            -1.1,
+            -1.01,
+            1.01,
+            1.1,
+            1.3,
+            1.7,
+            2.2,
+            2.6,
+            3.1,
+            3.7,
+            4.2,
+            5.0
```
```suggestion
            -5,
            -4.2,
            -3.7,
            -3.1,
            -2.6,
            -2.2,
            -1.7,
            -1.3,
            -1.1,
            -1.01,
            1.01,
            1.1,
            1.3,
            1.7,
            2.2,
            2.6,
            3.1,
            3.7,
            4.2,
            5
```

kgryte committed:
chore: remove trailing zeros

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/acscf/package.json:
```diff
@@ -66,4 +66,89 @@
+        "arccosecant",
+        "cosecant",
+        "inverse",
+        "trig",
+        "trigonometry",
+        "radians",
```
```suggestion
```

kgryte committed:
chore: remove keywords which are too general

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/acscf/package.json:
```diff
@@ -66,4 +66,86 @@
+        }
+      },
+      "keywords": [
+        "acscf",
```
```suggestion
        "acscf",
        "acsc",
```

kgryte committed:
docs: add keyword

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/acsch/package.json:
```diff
@@ -69,4 +69,92 @@
+            ]
+          },
+          "example_values": [
+            -5.0,
+            -3.7,
+            -2.9,
+            -2.1,
+            -1.7,
+            -1.3,
+            -1.1,
+            -0.7,
+            -0.51,
+            -0.25,
+            0.25,
+            0.51,
+            0.7,
+            1.1,
+            1.3,
+            1.7,
+            2.1,
+            2.9,
+            3.7,
+            5.0
```
```suggestion
            -5,
            -3.7,
            -2.9,
            -2.1,
            -1.7,
            -1.3,
            -1.1,
            -0.7,
            -0.51,
            -0.25,
            0.25,
            0.51,
            0.7,
            1.1,
            1.3,
            1.7,
            2.1,
            2.9,
            3.7,
            5
```

kgryte committed:
style: remove trailing zeros

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/acsch/package.json:
```diff
@@ -69,4 +69,92 @@
+        "csc",
+        "arc",
+        "arccosecant",
+        "trig",
+        "trigonometry",
```
```suggestion
```

kgryte committed:
chore: remove keywords which are too general

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/acsch/package.json:
```diff
@@ -69,4 +69,90 @@
+            "prng": "random/base/uniform",
+            "parameters": [
+              0.5,
+              5.0
```
```suggestion
              5
```

kgryte committed:
style: remove trailing zero

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/acsch/package.json:
```diff
@@ -69,4 +69,90 @@
+      "$schema": "math/base@v1.0",
+      "base_alias": "acsch",
+      "alias": "acsch",
+      "pkg_desc": "compute the hyperbolic arccosecant of a number",
+      "desc": "computes the hyperbolic arccosecant of a number",
```
```suggestion
      "pkg_desc": "compute the hyperbolic arccosecant of a double-precision floating-point number",
      "desc": "computes the hyperbolic arccosecant of a double-precision floating-point number",
```

kgryte committed:
docs: update descriptions

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asin/package.json:
```diff
@@ -69,4 +69,88 @@
+            0.85,
+            0.9,
+            0.95,
+            1.0
```
```suggestion
              -1,
              1
            ]
          },
          "example_values": [
            -1,
            -0.95,
            -0.9,
            -0.85,
            -0.75,
            -0.66,
            -0.5,
            -0.33,
            -0.25,
            -0.11,
            0,
            0.11,
            0.25,
            0.33,
            0.5,
            0.66,
            0.75,
            0.85,
            0.9,
            0.95,
            1
```

kgryte committed:
style: remove trailing zeros

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asin/package.json:
```diff
@@ -69,4 +69,88 @@
+        "arcsine",
+        "sine",
+        "inverse",
+        "trig",
+        "trigonometry",
+        "radians",
```
```suggestion
```

kgryte committed:
chore: remove keywords which are too general

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/package.json:
```diff
@@ -67,4 +67,86 @@
+          "rand": {
+            "prng": "random/base/uniform",
+            "parameters": [
+              -1.0,
+              1.0
+            ]
+          },
+          "example_values": [
+            -1.0,
+            -0.95,
+            -0.9,
+            -0.85,
+            -0.75,
+            -0.66,
+            -0.5,
+            -0.33,
+            -0.25,
+            -0.11,
+            0.0,
+            0.11,
+            0.25,
+            0.33,
+            0.5,
+            0.66,
+            0.75,
+            0.85,
+            0.9,
+            0.95,
+            1.0
```
```suggestion
              -1,
              1
            ]
          },
          "example_values": [
            -1,
            -0.95,
            -0.9,
            -0.85,
            -0.75,
            -0.66,
            -0.5,
            -0.33,
            -0.25,
            -0.11,
            0,
            0.11,
            0.25,
            0.33,
            0.5,
            0.66,
            0.75,
            0.85,
            0.9,
            0.95,
            1
```

kgryte committed:
style: remove trailing zeros

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/package.json:
```diff
@@ -67,4 +67,86 @@
+        "arcsine",
+        "sine",
+        "inverse",
+        "trig",
+        "trigonometry",
+        "radians",
```
```suggestion
```

kgryte committed:
chore: remove keywords which are too general

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/package.json:
```diff
@@ -67,4 +67,83 @@
+        }
+      },
+      "keywords": [
+        "asinf",
```
```suggestion
        "asinf",
        "asin",
```

kgryte committed:
chore: add keyword

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (dismissed):
LGTM

kgryte commented on lib/node_modules/@stdlib/math/base/special/acscf/package.json:
```diff
@@ -66,4 +66,87 @@
     "radians",
     "angle"
   ]
+,
```
```suggestion
  ],
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/acsch/package.json:
```diff
@@ -69,4 +69,90 @@
     "math.asinh",
     "asinh"
   ]
+,
```
```suggestion
  ],
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/asin/package.json:
```diff
@@ -69,4 +69,85 @@
     "polyfill",
     "ponyfill"
   ]
+,
```
```suggestion
  ],
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/asinf/package.json:
```diff
@@ -67,4 +67,84 @@
     "radians",
     "angle"
   ]
+,
```
```suggestion
  ],
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM after fixing formatting issues.
